src/types.ts

```diff
@@ -112,6 +123,37 @@ export interface PaletteColor {
   color: string | null;
 }
 
+interface TextAlign {
+  auto: boolean;
+  align: string;
+}
+
+export interface ExtendedNxAttrExprInfo extends EngineAPI.INxAttrExprInfo {
+  id: string;
+  qMin: number;
+  qMax: number;
+  qContinuousAxes: boolean;
+  qIsCyclic: boolean;
+  qFallbackTitle: string;
+}
+
+export interface ExtendedNxDimensionInfo extends Omit<EngineAPI.INxDimensionInfo, 'qAttrExprInfo'> {
```

Comment:
I've never seen this Omit syntax. isn't qAttrExprInfo: ExtendedNxAttrExprInfo[]; overriding it anyway?

Reply:
no, qAttrExprInfo already exists in EngineAPI.INxDimensionInfo. We can only extend (add) a new one.

Reply:
ok so you have to omit then add it back? otherwise you have a two props with the same name?

Reply:
Yes, otherwise typescript complains